sw/device/lib/base/BUILD

```diff
@@ -0,0 +1,50 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "bitfield",
+    linkstatic = True,
+    srcs = [
+        "bitfield.c",
+        "bitfield.h",
+        ],
+    )
+
+cc_library(
+    name = "mem",
+    linkstatic = True,
+    srcs = [
+        "memory.c",
+        "memory.h",
+        ],
+    copts = [
+        "-fno-builtin",
+        ],
+    )
+
+
+cc_library(
+    name = "mmio",
+    linkstatic = True,
+    srcs = [
+        "mmio.c",
+        "mmio.h",
+        ],
+    deps = [
+        ":bitfield",
+        ":mem",
+        ],
+    )
+
+cc_library(
+    name = "hardened",
+    linkstatic = True,
+    srcs = [
+        "hardened.c",
+        "hardened.h",
+        "stdasm.h",
+        ],
+    )
```

Comment:
You might want to group most of these together into a single target:
https://github.com/cfrantz/opentitan/blob/bazel/sw/device/lib/base/BUILD

Most modules are going to end up using several things from `base` and several of the libraries here are header-only (and we can rely on the linker to do the right thing with individual objects within a library).